cli/tools/repl.rs

```diff
@@ -328,171 +326,263 @@ Object.defineProperty(globalThis, "_error", {
 });
 "#;
 
-async fn inject_prelude(
-  worker: &mut MainWorker,
-  session: &mut LocalInspectorSession,
-  context_id: u64,
-) -> Result<(), AnyError> {
-  worker
-    .with_event_loop(
-      session
-        .post_message(
-          "Runtime.evaluate",
-          Some(json!({
-            "expression": PRELUDE,
-            "contextId": context_id,
-          })),
-        )
-        .boxed_local(),
-    )
-    .await?;
+struct ReplSession {
+  worker: MainWorker,
+  session: LocalInspectorSession,
+  pub context_id: u64,
+}
 
-  Ok(())
+impl ReplSession {
+  pub async fn initialize(mut worker: MainWorker) -> Result<Self, AnyError> {
+    let mut session = worker.create_inspector_session().await;
+
+    worker
+      .with_event_loop(
+        session.post_message("Runtime.enable", None).boxed_local(),
+      )
+      .await?;
+
+    // Enabling the runtime domain will always send trigger one executionContextCreated for each
+    // context the inspector knows about so we grab the execution context from that since
+    // our inspector does not support a default context (0 is an invalid context id).
+    let mut context_id: u64 = 0;
```

Comment:
The code here for getting the `context_id` has not changed as I tried not to change any logic in this PR, but it seems like it should be `session.notifications().find(|n| n.get("method").unwrap().as_str().unwrap() == "Runtime.executionContextCreated")` or whatever and then throw when it can't be found since `context_id == 0` is invalid (as per comment above)?

Reply:
It could be actually hardcoded to `1`. `deno_core` currently doesn't support more than one context and there are TODOs scattered around different parts of the codebase mentioning that. I'm fine with leaving as is.